User can login to application

jonas-masharqa opened the pull request:
[PT-Link](https://www.pivotaltracker.com/story/show/169667180)

## PR Contains
```
As a user of the web application
In order to access more personalised functionality
I would like to be able to log in to my account 
```

## PR Content
- Redux token auth package & wrapping of app
- Redux store, action and reducer files
- Integration test Happy & Sad path + fixture
- Login & LoginForm components + routes
- Message handling
- Redirect on successful sign in
- Refactoring

## Screenshots
<img width="1102" alt="Screenshot 2019-11-13 at 16 15 36" src="https://user-images.githubusercontent.com/53903972/68777145-830c3180-0631-11ea-800a-48593a7eee1c.png">

<img width="1440" alt="Screenshot 2019-11-13 at 16 14 44" src="https://user-images.githubusercontent.com/53903972/68777129-7ee01400-0631-11ea-8c91-fabf008d88c0.png">

<img width="742" alt="Screenshot 2019-11-13 at 16 16 09" src="https://user-images.githubusercontent.com/53903972/68777154-856e8b80-0631-11ea-963d-ce9e11dd5598.png">

lucamarial committed:
Installs react redux token auth packages and creates store, actions and reducers folders

jonas-masharqa committed:
Merge branch 'development' of github.com:CraftAcademy/wander-client into login

jonas-masharqa committed:
Creates integration test and fixture for user can login too application.

jonas-masharqa committed:
Creates Login and LoginForm components.

lucamarial committed:
Adds login route compoennt and NavLink, adds styling to LoginForm and inputChangeHandler to Login component

jonas-masharqa committed:
Wraps app in Provider and configures store.

piavW committed:
Creates and connects reduxTokenAuthConfig to login
- Creates actions folder, creates and configures reduxTokenAuthConfig file
- Creates user attributes in reduxTokenAuthConfig
- Creates reducers folder, creates and configures rootReducer file
- Connects Login to redux
- Adds username attribute to state and loginform

jonas-masharqa committed:
Corrects login functionality

- Creates handleLogin function
- Adds message handling
- Refactors LoginForm with correct id's and props
- Corrects AuthURL
- Corrects integration test
- Renames fixture file and adds username

Co-Authored-By: Pia von Wachenfelt, <piavonwachenfelt@hotmail.com>

piavW committed:
Creates fixture file and test for sadpath.

jonas-masharqa committed:
Refactors login functionality and user attributes

- Maps state to props in App
- Connects App
- Redirects user on successful login with ternary operator in routes
- Changes username to name in relevant files
- Moves welcomeMessage to be displayed LandingPage

lucamarial commented on src/Components/Login.jsx:
```diff
@@ -0,0 +1,72 @@
+class Login extends Component {
+  state = {
+    email: '',
+    username: '',
```
Do you use the username here?

jonas-masharqa replied:
Removed

piavW committed:
Replaces old syntax.

jonas-masharqa committed:
Removes username state from Login component.

GergKllai1 commented on cypress/integration/userCanLogin.spec.js:
```diff
@@ -0,0 +1,68 @@
+      method: 'GET',
+      url: 'http://localhost:3000/v1/trails',
+      response: 'fixture:user_can_view_list_of_trails.json'
+    })
```
You can put this and cy.server() into `cypress/support/index.js` file and then you don't have to call it in every test file/suite.

This is an example:

https://github.com/CraftAcademy/gnosis-client/blob/development/cypress/support/index.js

jonas-masharqa replied:
Done! Really cool thanks!

GergKllai1 commented on src/App.jsx:
```diff
@@ -1,19 +1,32 @@
+import Login from './Components/Login'
 
-const App = () => {
+const App = ({ currentUser }) => {
```
I like it :)

faraznaeem commented on src/Components/Login.jsx:
```diff
@@ -0,0 +1,71 @@
+class Login extends Component {
+  state = {
+    email: '',
+    password: '',
```
Why do you need to save the password in the state?

piavW replied:
Yes we need it to pass it into the handleLogin function. Is there another way? Are we missing something?

faraznaeem commented on src/Components/Login.jsx:
```diff
@@ -0,0 +1,71 @@
+    errorMessage: ''
+  }
+
+  inputChangeHandler = (e) => {
```
Please be consistent with your code formatting, make sure everyone uses prettier.

faraznaeem commented on src/Components/Login.jsx:
```diff
@@ -0,0 +1,71 @@
+    const { email, password } = this.state;
+      signInUser({ email, password })
+        .then(
+          console.log('Wander on..')
```
I don't understand why you are using `console.log` here?

piavW replied:
The .then requires to have something within it, and it's a good verification for us that signInUser-function was used.

GergKllai1 replied:
Usually this could be the place where you can redirect. If I see correctly you have the login on a separate route, so you can use the react router to navigate back to the index page for example by doing:
```
this.props.history.push('/')
```

piavW replied:
Cool! We'll do it!

faraznaeem commented on src/Components/Login.jsx:
```diff
@@ -0,0 +1,71 @@
+    let renderLogin, errorMessage
+
+    if (this.state.errorMessage) {
+      errorMessage = <p id="error-message">{this.state.errorMessage}</p>
```
is this neccesary? `this.state.errorMessage`

jonas-masharqa committed:
Refactoring

- Extracts beforeEach in integration tests to support
- Removes if statements from errorMessage(s)
- Updates code formatting to be consistent in Login

Co-Authored-By: Pia von Wachenfelt, <piavonwachenfelt@hotmail.com>
Co-Authored-By: David Izosimov, <david.izosimov@gmail.com>

piavW committed:
Refactors welcomeMessage and beautifies code.

piavW committed:
Changes welcomemessage in test.

piavW committed:
Removes beforeEach in integrationtest due to error in semiphore job.
